core/shared/src/main/scala/coursier/util/Tree.scala

```diff
@@ -16,8 +16,9 @@ object Tree {
       */
     def recursivePrint(elems: Seq[T], ancestors: Set[T], prefix: String, acc: String => Unit): Unit = {
       val unseenElems: Seq[T] = elems.filterNot(ancestors.contains)
-      for (elem <- unseenElems) {
-        val isLast = unseenElems.indexOf(elem) == unseenElems.length - 1
```

Comment:
I think the call to `indexOf` makes things quadratic here (calling `indexOf` for each element should be quadratic wrt to `unseenElems.length`). I pushed a quick fix.

Reply:
ok good catch. thanks!